code_editor/src/components/ACodeEditor.vue

```diff
@@ -0,0 +1,41 @@
+<template>
+	<div ref="container" id="container"></div>
+</template>
+
+<script setup lang="ts">
+import loader from '@monaco-editor/loader'
+import { editor } from 'monaco-editor'
+import { onMounted, ref } from 'vue'
+
+import { theme } from '@/theme/code_editor/agritheory'
+
+const props = defineProps<{
+	options?: editor.IStandaloneEditorConstructionOptions
+}>()
+
+const container = ref(null)
+const editorOptions: typeof props['options'] = {
+	...props.options,
+	automaticLayout: true,
+	colorDecorators: true,
+	lineHeight: 24,
+}
+
+onMounted(async () => {
+	const monacoInstance = await loader.init()
+	const editor = monacoInstance.editor
+
+	editor.defineTheme('agritheory', theme)
+	editor.setTheme('agritheory')
+
+	// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+	editor.create(container.value as HTMLElement, editorOptions)
+})
+</script>
+
+<style scoped>
+#container {
```

Comment:
I am really uncomfortable with this global id. Is this required by Monaco? Can we refactor it to a different selector?
Related, the height styling doesn't really make sense to me: when I change it to, for example, '30vh' in the story, there is still ~50% more code editor space that appears in on scroll. When I change it to '100vh' there is still more code editor area below the scroll.

Reply:
@agritheory, turns out Monaco will add some view height beyond the last line. We can either disable it for the component, or just for the story.

Reply:
@Alchez Let's disable it. It's going to scroll appropriately

Reply:
Alright, I've disabled it for the entire component in the last commit. If we want the user to allow for that, I can move it to the story instead.